lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -307,11 +307,14 @@ def json_error(raw_response)
       end
 
       def headers(action, post, authorization = nil)
-        {
+        headers = {
           'Content-Type' => content_type,
           'Authorization' => auth_digest(action, post, authorization),
           'Date' => date
         }
+
+        headers['x-gcs-idempotence-key'] = options[:idempotency_key] if options[:idempotency_key]
```

Comment:
This header key is lowercase due to the [example headers in Ingenico's documentation](https://epayments-api.developer-ingenico.com/s2sapi/v1/en_US/java/authentication.html?paymentPlatform=GLOBALCOLLECT), but I'm open to feedback about this 📝

Reply:
Thank you for linking the documentation.

My understanding is that HTTP headers are case-insensitive. So theoretically it shouldn't matter either way. 

That being said, I couldn't find any lowercase examples in the documentation, am I missing it? The only examples I can find of this header are from [this page](https://epayments-api.developer-ingenico.com/s2sapi/v1/en_US/java/idempotent-requests.html?paymentPlatform=GLOBALCOLLECT), and they show it being capitalized.

I agree with you that it would be preferable to follow whatever convention is displayed in the gateway documentation. If the docs show both uppercase and lowercase, then we might as well leave it as is. But if it's only showing uppercase then I would request you change it to be uppercase.

Reply:
You new revisions took me deeper into Ingenico's documentation on building their authorization signature, and you are right that the header needs to be lowercase, at least in the `auth_digest` method because they state that you must lowercase each header name when creating the signature.